packages/eslint-config-airbnb/rules/react.js

```diff
@@ -465,6 +466,14 @@ module.exports = {
     // https://github.com/yannickcr/eslint-plugin-react/blob/bc976b837abeab1dffd90ac6168b746a83fc83cc/docs/rules/jsx-fragments.md
     // TODO: enable, semver-major
     'react/jsx-fragments': ['off', 'syntax'],
+
+    // Enforce Rules of Hooks
+    // https://github.com/facebook/react/blob/f99fca3cb28e1fbd87e0b647ac43959add290be7/packages/eslint-plugin-react-hooks/README.md
+    'react-hooks/rules-of-hooks': 'error',
```

Comment:
let's add this in a separate entry point; i'm thinking `hooks.js`, so that you can extend `['airbnb', 'airbnb/hooks']`. That way, folks that aren't using hooks or don't want to, don't have to.

Reply:
Won't having two entry points add unnecessary complexity? If you do not use hooks, these rules won't affect you. But folks may start using hooks thinking they are covered while they're not.

Reply:
They very well might, since component detection is hard, and functions whose names start with “use” sent something hooks invented.

Reply:
Is it to avoid introducing an additional peer dependency? Or just that airbnb's position is that hooks are bad so you would rather ignore them by default?

Reply:
No, we'd still require the additional peer dependency regardless - there's only one package.json per package.

Airbnb definitely won't be recommending against using hooks; but that doesn't mean there's no caveats with them.

Reply:
That's what I figured. I suppose you could still not add the peer and document that it is required in this particular condition. But it looked like a terrible idea.
To be honest, I still don't quite understand why it is best to keep these rules as a separated config in this case. But thanks for the answer.

Reply:
> To be honest, I still don't quite understand why it is best to keep these rules as a separated config in this case.

I can see both sides of the argument, but I also see no harm in making them an opt-in. If that's not the case, the individual rules can still be disabled if people choose not to use hooks, but... that's just unnecessary noise and it requires people to maintain their config if more hook rules were added in the future.

>  If you do not use hooks, these rules won't affect you. But folks may start using hooks thinking they are covered while they're not.

I don't necessarily disagree, but worth mentioning that I found a couple of places where methods prefixed with `use` (that weren't hooks) in a codebase I'm working on. It's possibly an edge case, but it could happen! The rules only use the function name to decide if it is a hook.

For example, with `eslint-plugin-react-hooks` added, the following file will throw an error even though I'm not using React at all:

```js
function useSomething() {}

export function doSomething() {
  if (window.totallyNotAReactComponent) {
    useSomething();
    ~~~~~~~~~~~~
  }
}
```